chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java

```diff
@@ -310,21 +373,44 @@ public String deleteUser(@PathVariable("userId") Long userId)
         return "{\"deletedUser\" : true}";
     }
 
-    @ApiOperation(value = "Give additional roles to a user.",
+    @Deprecated
+    @ApiOperation(value = "DEPRECATED.  Give additional roles to a user.",
             notes = "Users may be given ROLE_ADMIN, ROLE_ACB, "
                     + "ROLE_ATL, or ROLE_ONC_STAFF roles within the system.")
     @RequestMapping(value = "/grant_role", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
             produces = "application/json; charset=utf-8")
-    public String grantUserRole(@RequestBody GrantRoleJSONObject grantRoleObj) throws InvalidArgumentsException,
+    public String grantUserRoleDeprecated(@RequestBody GrantRoleJSONObject grantRoleObj) throws InvalidArgumentsException,
             UserRetrievalException, UserManagementException, UserPermissionRetrievalException, JsonProcessingException,
             EntityCreationException, EntityRetrievalException {
+        
+        return grant(grantRoleObj);
+    }
+
+    @ApiOperation(value = "Give additional roles to a user.",
+            notes = "Users may be given ROLE_ADMIN, ROLE_ACB, "
+                    + "ROLE_ATL, or ROLE_ONC_STAFF roles within the system.")
+    @RequestMapping(value = "/{userName}/roles/{roleName}", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
```

Comment:
Isn't this an edit of a User also? So a PUT?

Reply:
This creates a relationship between a user and role, therefore I think it should be a POST.